sdk/core/azure-core/azure/core/pipeline/transport/_base.py

```diff
@@ -290,6 +295,13 @@ def _format_data(data):
     def format_parameters(self, params):
         # type: (Dict[str, str]) -> None
         """Format parameters into a valid query string.
+
+        We recommend to directly pass your query parameter
```

Comment:
I think it's too alarmist for nothing. Just tell people this method will be called automatically if they pass `query`, that's it. The behavior will be the same in the end anyway.